src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -353,7 +353,7 @@ class %s extends {$options['class']}
 EOF;
 
                 foreach ($this->preload as $class) {
-                    if (!$class || str_contains($class, '$') || \in_array($class, ['int', 'float', 'string', 'bool', 'resource', 'object', 'array', 'null', 'callable', 'iterable', 'mixed', 'void'], true)) {
+                    if (!$class || str_contains($class, '$') || \in_array($class, ['int', 'float', 'string', 'bool', 'resource', 'object', 'array', 'null', 'callable', 'iterable', 'mixed', 'void'], true) || \function_exists($class)) {
```

Comment:
I'd rather patch the getClasses method to better deal with factories, can you have a look?

Reply:
we could add 'never' to that list BTW :)

Reply:
@nicolas-grekas Should i just move `\function_exists` there? Or you do have other idea?